src/as-release.c

```diff
@@ -273,6 +285,9 @@ as_release_set_version (AsRelease *release, const gchar *version)
 gint
 as_release_vercmp (AsRelease *rel1, AsRelease *rel2)
 {
+	g_return_val_if_fail (AS_IS_RELEASE (rel1), 0);
+	g_return_val_if_fail (AS_IS_RELEASE (rel2), 0);
```

Comment:
Those two checks aren't needed, as `as_release_get_version` already verifies the input.

Reply:
It still helps the consumer to know that it is the content that they passed to `vercmp` that is wrong and also to know if it is `rel1` or `rel2`